pkg/commands/compute/init.go

```diff
@@ -812,7 +835,29 @@ func (c *InitCommand) FetchPackageTemplate(branch, tag string, archives []file.A
 	}
 	c.Globals.ErrLog.Add(err)
 
-	req, err := http.NewRequest("GET", c.CloneFrom, nil)
+	// If this isn't a local file path, it should be a URL.
+	u, err := url.Parse(c.CloneFrom)
+	if err != nil {
+		spinner.StopFailMessage(msg)
+		spinErr := spinner.StopFail()
+		if spinErr != nil {
+			return fmt.Errorf(text.SpinnerErrWrapper, spinErr, err)
+		}
+		return fmt.Errorf("could not read --from URL: %w", err)
+	}
+
+	// If given an opaque string, the scheme and host are typically
+	// empty and the string ends up in u.Path.
+	if u.Host == "" && u.Scheme == "" {
+		spinner.StopFailMessage(msg)
+		spinErr := spinner.StopFail()
+		if spinErr != nil {
+			return fmt.Errorf(text.SpinnerErrWrapper, spinErr, err)
+		}
+		return fmt.Errorf("--from url seems invalid: %s", c.CloneFrom)
+	}
+
+	req, err := http.NewRequest("GET", u.String(), nil)
```

Comment:
I'm interested to know why the extra code here. Was there an unclear error output from `http.NewRequest("GET", c.CloneFrom, nil)` that adding this extra logic was able to provide clarity for?

Reply:
Ah that's the comment on 849/850. If you pass something that's _almost_ a service ID but is too short or too long into `url.Parse`, you'll get something like

```go
u, _ := url.Parse("abcd1234")
fmt.Printf("%#v\n", u")
// url.URL{
//   Scheme: "",
//   Host:   "",
//   Path:   "abcd1234",
    // ... etc
// }
```

Without this check, the errors trying to fetch the path with no other information were pretty obtuse because they came out of the HTTP client. How could I make this clearer?

Reply:
No that's fine, that makes sense now 👍🏻